src/main/savestates.c

```diff
@@ -839,6 +839,18 @@ static int savestates_load_m64p(struct device* dev, char *filepath)
             curr += sizeof(uint32_t);
 #endif
         }
+
+        if (version >= 0x00010700)
+        {
+            dev->sp.fifo[0].dir = GETDATA(curr, uint32_t);
+            dev->sp.fifo[0].length = GETDATA(curr, uint32_t);
+            dev->sp.fifo[0].memaddr = GETDATA(curr, uint32_t);
+            dev->sp.fifo[0].dramaddr = GETDATA(curr, uint32_t);
+            dev->sp.fifo[1].dir = GETDATA(curr, uint32_t);
+            dev->sp.fifo[1].length = GETDATA(curr, uint32_t);
+            dev->sp.fifo[1].memaddr = GETDATA(curr, uint32_t);
+            dev->sp.fifo[1].dramaddr = GETDATA(curr, uint32_t);
+        }
```

Comment:
Still, you might want to explicitely clear DMA_BUSY/DMA_FULL flags to make sure fifo content is not accessed if version is < 1.7